### PR TITLE
Make decloaking caused by insufficient fuel faster than normal decloaking

### PR DESCRIPTION
**Pull request by tehhowch**
This PR addresses #2557 (erratic AI targeting of persistently-almost-cloaked ships) by increasing the time between becoming targetable (`cloak < 1`) and not targetable (`cloak = 1`).

Current game mechanics allow IsTargetable to toggle as often as every frame, which causes AI targeting and firing behavior to appear erratic and seemingly broken. If instead a forced decloak (due to insufficient fuel or energy for the cloaking device) causes the cloak to drop by an additional amount (i.e. 10% cloak), there is a twofold effect:
1) The system radius in which the player can elicit the "erratic" targeting behavior is significantly reduced, due to the change in fuel generation (11 out of 12 frames must be fully powered & fueled instead of 1 out of 2)
2) A minimum of 11 frames will occur between the decloaked ship becoming targetable and then non-targetable, so the erratic targeting - even when active - is not as fast.

TLDR: instead of 
`cloak = 1 -> cloak = .99 -> cloak = 1 -> cloak = .99 ...`
I propose
`cloak = 1 -> cloak = .89 -> cloak = .90 -> ... -> cloak = .99 -> cloak = 1 -> cloak = .89 ...`

### CODE REVIEW

**Comment by endless-sky** [comment]
That feels like a rather clumsy way to solve this. It might make more sense to instead say that a ship can't start cloaking again until it is done uncloaking, or something like that.

**Comment by tehhowch** [comment]
I chose this method as it doesn't remove the existing strategy of dealing with overwhelming missile numbers, which is to cloak and then decloak just long enough to get the AI to fire off Nuclear Missiles or huge swarms of Hai Trackers, and then re-cloak before they reach your ship.

If one can only initiate cloaking after fully decloaking, the Reconciliation missions dealing with NMs may need rebalancing.

**Comment by gunqqer** [comment]
I agree with tehhowch, the only reason why I survive the nukes in my clipper was because of quick decloak-cloak. If the decloaking was voluntary you should be able to instantly reactivate it (so if I'm half way cloaked while uncloaking, I go up from 50%) 

If I'm involuntarily being uncloaked via lack of energy/fuel, or possible new tech that may hopeful you come in the future I should have to wait till 0% cloak to re-initiate cloaking.

**Comment by endless-sky** [comment]
Well, my idea was to only require the cool-down if the reason for decloaking was running out of fuel (added in d754ba6).

The additional benefit of doing it that way is that eventually, there might be other things that force a ship into a decloaking cycle or prevent it from cloaking in the first place (i.e. getting hit by certain weapons). Since blast radius weapons hit even cloaked ships, that would allow blast weapons that force every cloaked ship in the area to decloak for some amount of time.

**Comment by tehhowch** [comment]
Yes, that's a nice way of doing it, and the forced full de-cloak makes the targeting behavior smooth as well.
